src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -1694,4 +1694,172 @@ public void testUpdateLocatorsCopyUnsupported() throws SQLException {
+    assertEquals(0, getSizeOfResultSet(dbmd.getPrimaryKeys(database, schemaPattern, table1)));
+
+    // Should return an empty result if we try a pattern match on the table name
+    assertEquals(0, getSizeOfResultSet(dbmd.getPrimaryKeys(database, null, "T%")));
+
+    // Should return result for matching schema and table name
+    assertEquals(1, getSizeOfResultSet(dbmd.getImportedKeys(database, schema, table2)));
+
+    // Should return an empty result if we try a pattern match on the schema
+    assertEquals(0, getSizeOfResultSet(dbmd.getImportedKeys(database, schemaPattern, table2)));
+
+    // Should return an empty result if we try a pattern match on the table name
+    assertEquals(0, getSizeOfResultSet(dbmd.getImportedKeys(database, null, "T%")));
+
+    // Should return result for matching schema and table name
+    assertEquals(1, getSizeOfResultSet(dbmd.getExportedKeys(database, schema, table1)));
+
+    // Should return an empty result if we try a pattern match on the schema
+    assertEquals(0, getSizeOfResultSet(dbmd.getExportedKeys(database, schemaPattern, table1)));
+
+    // Should return an empty result if we try a pattern match on the table name
+    assertEquals(0, getSizeOfResultSet(dbmd.getExportedKeys(database, null, "T%")));
+
+    // Should return result for matching schema and table name
+    assertEquals(
+        1,
+        getSizeOfResultSet(
+            dbmd.getCrossReference(database, schema, table1, database, schema, table2)));
+
+    // Should return an empty result if we try a pattern match on any of the table or schema names
+    assertEquals(
+        0,
+        getSizeOfResultSet(
+            dbmd.getCrossReference(database, schemaPattern, table1, database, schema, table2)));
+    assertEquals(
+        0,
+        getSizeOfResultSet(
+            dbmd.getCrossReference(database, schema, table1, database, schemaPattern, table2)));
+    assertEquals(
+        0,
+        getSizeOfResultSet(dbmd.getCrossReference(database, null, "T%", database, schema, table2)));
+    assertEquals(
+        0,
+        getSizeOfResultSet(dbmd.getCrossReference(database, schema, table1, database, null, "T%")));
+  }
+
+  @Test
+  public void testPatternSearchAllowedForPrimaryAndForeignKeys() throws SQLException {
+    Map<String, String> params = getConnectionParameters();
+    Properties properties = new Properties();
+    for (Map.Entry<?, ?> entry : params.entrySet()) {
+      if (entry.getValue() != null) {
+        properties.put(entry.getKey(), entry.getValue());
+      }
+    }
+    // test out connection parameter stringsQuoted to remove strings from quotes
+    properties.put("enablePatternSearch", "true");
+    Connection connection = DriverManager.getConnection(params.get("uri"), properties);
+
+    String database = connection.getCatalog();
+    String schema = connection.getSchema();
+    String schemaPattern = schema.substring(0, 1).concat("%");
```

Comment:
we are taking only the first character? will it be stable when we run many tests concurrently on the same database?